module/optee/smt/test/mod_optee_smt_unit_test.c

```diff
@@ -0,0 +1,889 @@
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, channel_ctx->max_payload_size, (const void*)fake_payload, channel_ctx->max_payload_size);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* No message available */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_write_payload(channel_id, 0, (const void*)fake_payload, channel_ctx->max_payload_size);
+    TEST_ASSERT_EQUAL(FWK_E_ACCESS, status);
+}
+
+/*!
+ * \brief test write_payload function.
+ *
+ *  \details Test all remaining cases for write_payload function.
+ */
+void test_smt_write_payload(void)
+{
+    int status;
+    fwk_id_t channel_id;
+    char fake_payload[FAKE_MAILBOX_SIZE];
+    struct smt_channel_ctx *channel_ctx;
+
+    /* Init channel context */
+    channel_ctx = &smt_ctx.channel_ctx_table[FAKE_SCMI_SERVICE_IDX_OSPM_0];
+
+    /* write payload */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_str_memcpy_StubWithCallback(fake_fwk_str_memcpy);
+    status = smt_write_payload(channel_id, 0, (const void*)fake_payload, channel_ctx->max_payload_size);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+/*!
+ * \brief test early access to signal_message.
+ *
+ *  \details Test all cases for signaling a message to a static memory.
+ */
+void test_smt_signal_message_early(void)
+{
+    int status;
+    fwk_id_t channel_id;
+
+   /* Mailbox not ready, discard message silently ... */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_signal_message(channel_id);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+}
+
+static int fake_signal_message(fwk_id_t service_id)
+{
+    return fake_signal_message_status;
+}
+
+struct mod_scmi_from_transport_api fake_scmi_api = {
+    .signal_error = NULL, /* Not used */
+    .signal_message = fake_signal_message,
+};
+
+static int fake_raise_smt_interrupt(fwk_id_t channel_id)
+{
+    return FWK_SUCCESS;
+}
+
+struct mod_optee_smt_driver_api fake_mbx_smt_api = {
+    .raise_interrupt = fake_raise_smt_interrupt,
+};
+
+void test_reset_virt_buffer(void)
+{
+    int i;
+
+    for (i = 0; i < FAKE_MAILBOX_SIZE; i++) {
+        mailbox_virt_buffer[i] = 0xAA;
+    }
+}
+
+void test_reset_in_buffer(void)
+{
+    int i;
+
+    for (i = 0; i < FAKE_MAILBOX_SIZE; i++) {
+        channel_in_buffer[i] = 0;
+    }
+}
+
+void test_reset_out_buffer(void)
+{
+    int i;
+
+    for (i = 0; i < FAKE_MAILBOX_SIZE; i++) {
+        channel_out_buffer[i] = 0;
+    }
+}
```

Comment:
I suggest for these helper functions to remove the "test" prefix
to not conflict with the real tests.